### PR TITLE
Added recommendation to set pool.min to 0

### PR DESCRIPTION
**Pull request by mjomble**
This is a new version of PR #393 which became outdated during the migration to VitePress

### CODE REVIEW

**Comment by kibertoad** [comment]
@mjomble Can you rephrase it to be closer to https://github.com/Vincit/tarn.js/pull/74/files? From my understanding this is only preferable if it is important to destroy stale inactive connections, some users probably may want to keep some connections running in the pool?

@elhigu please correct me if I'm wrong

**Comment by mjomble** [comment]
The original PR was based on @elhigu's comment on https://github.com/knex/knex/issues/4525#issuecomment-862497867

> I agree that 0 would be better default for it. It is 2 because of historical reasons when different pools were used, which also did behave in different manner when initialised for the first time.

So my understanding is that ideally 0 would be the default, but it has to be 2 for backwards compatibility.

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Got it, thank you!
